projects/boost_headers.cmake

```diff
@@ -1,8 +1,12 @@
-ExternalProject_Add(BoostHeaders
-    URL http://sourceforge.net/projects/boost/files/boost/1.67.0/boost_1_67_0.tar.bz2
-    URL_HASH SHA1=694ae3f4f899d1a80eb7a3b31b33be73c423c1ae
-    CONFIGURE_COMMAND ""
-    BUILD_COMMAND ""
-    INSTALL_COMMAND "${CMAKE_COMMAND}" -E copy_directory "${CMAKE_CURRENT_BINARY_DIR}/BoostHeaders-prefix/src/BoostHeaders/boost" "${CMAKE_INSTALL_PREFIX}/include/boost"
-)
+#Copyright (c) 2020 Ultimaker B.V.
+#cura-build-environment is released under the terms of the AGPLv3 or higher.
 
+if(NOT TARGET BoostHeaders)
+    ExternalProject_Add(BoostHeaders
+        URL http://sourceforge.net/projects/boost/files/boost/1.67.0/boost_1_67_0.tar.bz2
+        URL_HASH SHA1=694ae3f4f899d1a80eb7a3b31b33be73c423c1ae
+        CONFIGURE_COMMAND ""
+        BUILD_COMMAND ""
+        INSTALL_COMMAND "${CMAKE_COMMAND}" -E copy_directory "${CMAKE_CURRENT_BINARY_DIR}/BoostHeaders-prefix/src/BoostHeaders/boost" "${CMAKE_INSTALL_PREFIX}/include/boost"
+    )
+endif()
```

Comment:
no newline

Reply:
You're right, newline isn't required, but considered good practice. Fine by me if stays like this.

(semi-)off-topic: I knew that it is in the PEP8 for Python but I didn't know it was also specified for C and C++

>Note that it is a good style to always put the newline as a last character if it is allowed by the file format. Furthermore, for example, for C and C++ header files it is required by the language standard.

Source: https://stackoverflow.com/a/5813359/1473298